src/org/exist/storage/io/AbstractVariableByteInput.java

```diff
@@ -142,36 +143,35 @@ public int read(byte b[], int off, int len) throws IOException {
         return i;
     }
 
-    public void copyTo(VariableByteOutputStream os) throws IOException {
+    public void copyTo(final VariableByteOutputStream os) throws IOException {
         int more;
         do {
             more = read();
-            os.buf.append((byte) more);
+            os.write((byte) more);
             more &= 0200;
         } while (more > 0);
     }
 
-    public void copyTo(VariableByteOutputStream os, int count)
+    @Override
+    public void copyTo(final VariableByteOutputStream os, final int count)
             throws IOException {
         int more;
         for (int i = 0; i < count; i++) {
             do {
                 more = read();
-                os.buf.append((byte)more);
+                os.write((byte)more);
                 more &= 0200;
             } while (more > 0);
         }
     }
-    
-    public void copyRaw(VariableByteOutputStream os, int count) throws IOException {
-        for (int i = 0; i < count; i++) {
-            os.buf.append((byte) read());
-        }
+
+    @Override
+    public void copyRaw(final VariableByteOutputStream os, final int count) throws IOException {
+        final byte buf[] = new byte[count];
+        read(buf, 0, count);
```

Comment:
`read` can read less bytes, so `write` must be called with len returned by `read`, no?